docs/notes/migration_guide.rst

```diff
@@ -0,0 +1,491 @@
+
+Review your code for any APIs marked with the ``FASTDDS_DEPRECATED`` and ``FASTDDS_TODO_BEFORE`` macros.
+Note that these deprecated APIs have been removed in Fast DDS v3.
+Make the necessary updates to your implementation to ensure compatibility with the new version.
+
+.. _step-8-update-structs-enums-and-variables:
+
+Step 8: Update Structs, Enums, and Variables
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+As part of the Fast DDS migration, several structs, enums, and variables have been updated. You will need to modify your code to reflect these changes:
+
+1. Enum and Variable Changes:
+
+    * ``DiscoveryProtocol_t`` is now ``DiscoveryProtocol``.
+    * Rename ``initialHeartbeatDelay`` to ``initial_heartbeat_delay``.
+    * Rename ``heartbeatPeriod`` to ``heartbeat_period``.
+    * Rename ``nackResponseDelay`` to ``nack_response_delay``.
+    * Rename ``nackSupressionDuration`` to ``nack_supression_duration``.
+    * Rename ``heartbeatResponseDelay`` to ``heartbeat_response_delay``.
+    * Rename ``initialAcknackDelay`` to ``initial_acknack_delay``.
+    * Rename ``expectsInlineQos`` to ``expects_inline_qos``.
+    * Rename ``m_typeSize`` to ``max_serialized_type_size``.
+    * Rename ``m_isGetKeyDefined`` to ``is_compute_key_provided``.
+    * Rename ``m_topicDataTypeName`` to ``topic_data_typename``.
+
+2. Extend Built-in Topics:
+
+    * ``SubscriptionBuiltinTopicData`` has been extended with additional fields to mimic those of ``ReaderProxyData``.
+    * ``PublicationBuiltinTopicData`` has been extended with additional fields to mimic those of ``WriterProxyData``.
+    * ``ParticipantBuiltinTopicData`` has been extended to include the product version and fields from ``ParticipantProxyData``.
+
+3. Other Struct Changes:
+
+    * ``SendBuffersAllocationAttributes`` has a new attribute to define the allocation configuration of the ``NetworkBuffers``.
+    * ``TypeConsistencyQos`` has been removed from ``DataReader``, and the ``TypeConsistencyEnforcementQosPolicy`` and ``DataRepresentationQosPolicy`` have been added.
+
+.. _step-9-refactor-examples:
+
+Step 9: Refactor Examples
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+All examples have been refactored to follow a consistent structure across the Fast DDS project.
+This includes renaming files, restructuring classes, and updating the overall format.
+Additionally, it is important to note that some examples have been removed, renamed, or had significant changes to their options and configurations.
+If you have integrated any of these examples into your own implementation, carefully review the updated examples to ensure compatibility with your project.
+
+All the examples have been refactored to follow the same structure:
+
+* File names, guards, and classes follow new format.
+* Detailed and well-formed README.md with example explanation.
+* Example structured in applications, stopped by `SIGTERM` signal.
+
+Hello World
+"""""""""""
```

Comment:
What do you think about linking the examples here? Apply in the remain examples if considered

```suggestion
`Hello World <https://github.com/eProsima/Fast-DDS/tree/master/examples/cpp/hello_world>`_
""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
```